akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -311,7 +311,16 @@ private[stream] class VirtualSubscriber[T](val owner: VirtualPublisher[T]) exten
  */
 private[stream] class VirtualPublisher[T]() extends Publisher[T] {
   @volatile var realPublisher: Publisher[T] = null
-  override def subscribe(s: Subscriber[_ >: T]): Unit = realPublisher.subscribe(s)
+  override def subscribe(s: Subscriber[_ >: T]): Unit = {
+    val sub = realPublisher.subscribe(s)
+    realPublisher match {
+      case _: SupportingMultipleSubscribers   ⇒
```

Comment:
Even if you go for "consistently single-subscriber" you should not remove the feature from fanoutPublisher since that is the whole point of that Sink. I guess the tests fail because they rely on the fact that the exposed publisher supports the multi-subscriber scenario the original publisher provides.